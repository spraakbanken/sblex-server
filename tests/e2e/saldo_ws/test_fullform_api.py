import pytest
from fastapi import status
from httpx import AsyncClient


class TestFullformRoutes:
    @pytest.mark.parametrize("fragment", ["dväljs", "dv", "dvä", "dväl"])
    @pytest.mark.asyncio
    async def test_json_valid_input_returns_200(
        self, client: AsyncClient, fragment: str, snapshot_json
    ):
        res = await client.get(f"/ff/json/{fragment}")
        assert res.status_code == status.HTTP_200_OK
        assert res.json() == snapshot_json

    @pytest.mark.parametrize("fragment", ["dväljs", "dv", "dvä", "dväl"])
    @pytest.mark.asyncio
    async def test_html_valid_input_returns_200(
        self, client: AsyncClient, fragment: str, snapshot
    ):
        res = await client.get(f"/ff/html?fragment={fragment}")
        assert res.status_code == status.HTTP_200_OK
        assert res.headers["content-type"] == "text/html; charset=utf-8"
        assert res.text == snapshot

    @pytest.mark.parametrize("fragment", ["dväljs", "dv", "dvä", "dväl"])
    @pytest.mark.asyncio
    async def test_html_orig_valid_input_returns_307(
        self,
        client: AsyncClient,
        fragment: str,
    ):
        res = await client.get(f"/ff/html/{fragment}")
        assert res.status_code == status.HTTP_307_TEMPORARY_REDIRECT

    @pytest.mark.parametrize("fragment", ["dväljs", "dv", "dvä", "dväl"])
    @pytest.mark.asyncio
    async def test_xml_valid_input_returns_200(
        self, client: AsyncClient, fragment: str, snapshot
    ):
        res = await client.get(f"/ff/xml/{fragment}")
        assert res.status_code == status.HTTP_200_OK
        assert res.headers["content-type"] == "application/xml"
        assert res.text == snapshot
