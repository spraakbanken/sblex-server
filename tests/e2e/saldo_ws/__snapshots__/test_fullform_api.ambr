# serializer version: 1
# name: TestFullformRoutes.test_html_valid_input_returns_200[dv\xe4]
  '''
  <!DOCTYPE html>
  <html>
  
  <head>
    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
    <link rel="shortcut icon" href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png">
    <link rel="stylesheet" type="text/css" href="http://testserver.saldo_ws/static/saldo.css" />
    <title>Fullform | dvä</title>
    
    <script src="https://unpkg.com/htmx.org@1.9.8" crossorigin="anonymous"></script>
  </head>
  
  <body>
    <center>
      <p>
        <a href="http://testserver.saldo_ws/fl/html"><img
            src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif" align="top"
            alt="SALDO: gå tillbaka till startsidan" /></a>
      </p>
    </center>
    <div id="output_table">
      
  <center>
      <form action="http://testserver.saldo_ws/ff/html" method="get" class="inputclass">
          <input type="search" id="search" class="inputclass" value="dvä" name="fragment" size="30"
              placeholder="Skriv in en ordform" results="10" />
          <input type="submit" value="Sök" />
      </form>
      
      <p>segment: dvä</p>
      <p>continuations: l </p>
      <table border="1">
          <tr>
              <th>gf</th>
              <th>id</th>
              <th>pos</th>
              <th>is</th>
              <th>msd</th>
              <th>p</th>
          </tr>
          
          
  </center>
  
    </div>
    <footer>
      <hr>
      <ul>
        <li><a href="https://spraakbanken.gu.se/saldo">SALDOs projektsida</a></li>
        <li><a href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf">dokumentation (pdf)</a></li>
      </ul>
    </footer>
  </body>
  
  </html>
  '''
# ---
# name: TestFullformRoutes.test_html_valid_input_returns_200[dv\xe4l]
  '''
  <!DOCTYPE html>
  <html>
  
  <head>
    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
    <link rel="shortcut icon" href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png">
    <link rel="stylesheet" type="text/css" href="http://testserver.saldo_ws/static/saldo.css" />
    <title>Fullform | dväl</title>
    
    <script src="https://unpkg.com/htmx.org@1.9.8" crossorigin="anonymous"></script>
  </head>
  
  <body>
    <center>
      <p>
        <a href="http://testserver.saldo_ws/fl/html"><img
            src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif" align="top"
            alt="SALDO: gå tillbaka till startsidan" /></a>
      </p>
    </center>
    <div id="output_table">
      
  <center>
      <form action="http://testserver.saldo_ws/ff/html" method="get" class="inputclass">
          <input type="search" id="search" class="inputclass" value="dväl" name="fragment" size="30"
              placeholder="Skriv in en ordform" results="10" />
          <input type="submit" value="Sök" />
      </form>
      
      <p>segment: dväl</p>
      <p>continuations: jt </p>
      <table border="1">
          <tr>
              <th>gf</th>
              <th>id</th>
              <th>pos</th>
              <th>is</th>
              <th>msd</th>
              <th>p</th>
          </tr>
          
          
  </center>
  
    </div>
    <footer>
      <hr>
      <ul>
        <li><a href="https://spraakbanken.gu.se/saldo">SALDOs projektsida</a></li>
        <li><a href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf">dokumentation (pdf)</a></li>
      </ul>
    </footer>
  </body>
  
  </html>
  '''
# ---
# name: TestFullformRoutes.test_html_valid_input_returns_200[dv\xe4ljs]
  '''
  <!DOCTYPE html>
  <html>
  
  <head>
    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
    <link rel="shortcut icon" href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png">
    <link rel="stylesheet" type="text/css" href="http://testserver.saldo_ws/static/saldo.css" />
    <title>Fullform | dväljs</title>
    
    <script src="https://unpkg.com/htmx.org@1.9.8" crossorigin="anonymous"></script>
  </head>
  
  <body>
    <center>
      <p>
        <a href="http://testserver.saldo_ws/fl/html"><img
            src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif" align="top"
            alt="SALDO: gå tillbaka till startsidan" /></a>
      </p>
    </center>
    <div id="output_table">
      
  <center>
      <form action="http://testserver.saldo_ws/ff/html" method="get" class="inputclass">
          <input type="search" id="search" class="inputclass" value="dväljs" name="fragment" size="30"
              placeholder="Skriv in en ordform" results="10" />
          <input type="submit" value="Sök" />
      </form>
      
      <p>segment: dväljs</p>
      <p>continuations:  </p>
      <table border="1">
          <tr>
              <th>gf</th>
              <th>id</th>
              <th>pos</th>
              <th>is</th>
              <th>msd</th>
              <th>p</th>
          </tr>
          
          <tr>
              <td>dväljas</td>
              <td><a href="http://testserver.saldo_ws/lid/html/dväljas..vb.1">dväljas..vb.1</a></td>
              <td>vb</td>
              <td></td>
              <td>pres ind s-form</td>
              <td>vb_vs_dväljas</td>
          </tr>
          
          <tr>
              <td>dväljas</td>
              <td><a href="http://testserver.saldo_ws/lid/html/dväljas..vb.1">dväljas..vb.1</a></td>
              <td>vb</td>
              <td></td>
              <td>imper</td>
              <td>vb_vs_dväljas</td>
          </tr>
          
          
  </center>
  
    </div>
    <footer>
      <hr>
      <ul>
        <li><a href="https://spraakbanken.gu.se/saldo">SALDOs projektsida</a></li>
        <li><a href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf">dokumentation (pdf)</a></li>
      </ul>
    </footer>
  </body>
  
  </html>
  '''
# ---
# name: TestFullformRoutes.test_html_valid_input_returns_200[dv]
  '''
  <!DOCTYPE html>
  <html>
  
  <head>
    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
    <link rel="shortcut icon" href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png">
    <link rel="stylesheet" type="text/css" href="http://testserver.saldo_ws/static/saldo.css" />
    <title>Fullform | dv</title>
    
    <script src="https://unpkg.com/htmx.org@1.9.8" crossorigin="anonymous"></script>
  </head>
  
  <body>
    <center>
      <p>
        <a href="http://testserver.saldo_ws/fl/html"><img
            src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif" align="top"
            alt="SALDO: gå tillbaka till startsidan" /></a>
      </p>
    </center>
    <div id="output_table">
      
  <center>
      <form action="http://testserver.saldo_ws/ff/html" method="get" class="inputclass">
          <input type="search" id="search" class="inputclass" value="dv" name="fragment" size="30"
              placeholder="Skriv in en ordform" results="10" />
          <input type="submit" value="Sök" />
      </form>
      
      <p>segment: dv</p>
      <p>continuations: äa </p>
      <table border="1">
          <tr>
              <th>gf</th>
              <th>id</th>
              <th>pos</th>
              <th>is</th>
              <th>msd</th>
              <th>p</th>
          </tr>
          
          
  </center>
  
    </div>
    <footer>
      <hr>
      <ul>
        <li><a href="https://spraakbanken.gu.se/saldo">SALDOs projektsida</a></li>
        <li><a href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf">dokumentation (pdf)</a></li>
      </ul>
    </footer>
  </body>
  
  </html>
  '''
# ---
# name: TestFullformRoutes.test_xml_valid_input_returns_200[dv\xe4]
  '''
  <?xml version="1.0" encoding="UTF-8"?>
  <result>
      <c>l</c>
      <as>
  
      </as>
  </result>
  '''
# ---
# name: TestFullformRoutes.test_xml_valid_input_returns_200[dv\xe4l]
  '''
  <?xml version="1.0" encoding="UTF-8"?>
  <result>
      <c>jt</c>
      <as>
  
      </as>
  </result>
  '''
# ---
# name: TestFullformRoutes.test_xml_valid_input_returns_200[dv\xe4ljs]
  '''
  <?xml version="1.0" encoding="UTF-8"?>
  <result>
      <c></c>
      <as>
  
          <a>
          <gf>dväljas</gf>
          <id>dväljas..vb.1</id>
          <pos>vb</pos>
          <is></is>
          <msd>pres ind s-form</msd>
          <p>vb_vs_dväljas</p>
          </a>
  
          <a>
          <gf>dväljas</gf>
          <id>dväljas..vb.1</id>
          <pos>vb</pos>
          <is></is>
          <msd>imper</msd>
          <p>vb_vs_dväljas</p>
          </a>
  
      </as>
  </result>
  '''
# ---
# name: TestFullformRoutes.test_xml_valid_input_returns_200[dv]
  '''
  <?xml version="1.0" encoding="UTF-8"?>
  <result>
      <c>äa</c>
      <as>
  
      </as>
  </result>
  '''
# ---
